File: Sources/Claudario/Mascot/MascotActivity.swift
import Foundation

enum MascotActivity: String, CaseIterable {
    case idle
    case thinking
    case reading
    case coding
    case running
    case planning
    case browsing
    case deepThink
    case compacting
    case dancing

    /// Emoji shown above the mascot's head while this activity is active.
    /// Empty string = no prop.
    var prop: String {
        switch self {
        case .idle:       return ""
        case .thinking:   return "❓"
        case .reading:    return "📖"
        case .coding:     return "⌨️"
        case .running:    return "⚡"
        case .planning:   return "📋"
        case .browsing:   return "🌐"
        case .deepThink:  return "💭"
        case .compacting: return "🌀"
        case .dancing:    return "🎉"
        }
    }

    /// Multiplier applied to base walkSpeed when state is .walking.
    var speedMultiplier: CGFloat {
        switch self {
        case .running:   return 1.6
        case .browsing:  return 0.7
        case .deepThink: return 0.5
        case .compacting: return 0  // motion handled separately (rotate)
        case .dancing:   return 0   // x-jumps, not walking
        default:         return 1.0
        }
    }

    /// Maps a Claude Code tool_name to the activity category. Returns
    /// `.thinking` for unknown / future tools.
    static func category(forTool name: String) -> MascotActivity {
        switch name {
        case "Read", "Grep", "Glob", "NotebookRead":
            return .reading
        case "Edit", "Write", "MultiEdit", "NotebookEdit":
            return .coding
        case "Bash", "BashOutput", "KillShell":
            return .running
        case "WebFetch", "WebSearch":
            return .browsing
        case "TodoWrite", "ExitPlanMode":
            return .planning
        case "Task":
            return .deepThink
        default:
            return .thinking
        }
    }
}

File: windows/Claudario.Windows/Mascot/MascotActivity.cs
namespace Claudario.Windows.Mascot;

public enum MascotActivity
{
    Idle, Thinking, Reading, Coding, Running,
    Planning, Browsing, DeepThink, Compacting, Dancing
}

public static class MascotActivityExtensions
{
    // Matches macOS MascotActivity.prop
    public static string Prop(this MascotActivity a) => a switch
    {
        MascotActivity.Thinking   => "❓",
        MascotActivity.Reading    => "📖",
        MascotActivity.Coding     => "⌨️",
        MascotActivity.Running    => "⚡",
        MascotActivity.Planning   => "📋",
        MascotActivity.Browsing   => "🌐",
        MascotActivity.DeepThink  => "💭",
        MascotActivity.Compacting => "🌀",
        MascotActivity.Dancing    => "🎉",
        _                         => "",
    };

    // Matches macOS MascotActivity.speedMultiplier
    public static double SpeedMultiplier(this MascotActivity a) => a switch
    {
        MascotActivity.Running    => 1.6,
        MascotActivity.Browsing   => 0.7,
        MascotActivity.DeepThink  => 0.5,
        MascotActivity.Compacting => 0,   // rotation, not walking
        MascotActivity.Dancing    => 0,   // x-jumps, not walking
        _                         => 1.0,
    };

    // Matches macOS MascotActivity.category(forTool:)
    public static MascotActivity CategoryForTool(string toolName) => toolName switch
    {
        "Read" or "Grep" or "Glob" or "NotebookRead"           => MascotActivity.Reading,
        "Edit" or "Write" or "MultiEdit" or "NotebookEdit"     => MascotActivity.Coding,
        "Bash" or "BashOutput" or "KillShell"                  => MascotActivity.Running,
        "WebFetch" or "WebSearch"                              => MascotActivity.Browsing,
        "TodoWrite" or "ExitPlanMode"                          => MascotActivity.Planning,
        "Task"                                                 => MascotActivity.DeepThink,
        _                                                      => MascotActivity.Thinking,
    };
}
